lpcompiler.pas

```diff
@@ -1762,6 +1780,23 @@ function TLapeCompiler.ParseMethod(FuncForwards: TLapeFuncForwards; FuncHeader:
       if (Result.Statements = nil) or (not (Result.Statements.Statements[Result.Statements.Statements.Count - 1] is TLapeTree_StatementList)) then
         Expect(tk_kw_Begin, False, False);
 
+      if (lcoHints in FOptions) then
+        with Result.StackInfo do
+          for i := 0 to VarCount - 1 do
+          begin
+            if (Vars[i].Used) or (Vars[i].Name = '') or (Vars[i].Name = 'Self') or (Vars[i].Name[1] = '!') then
+              Continue;
+
+            if (Vars[i] is TLapeParameterVar) then
+            begin
+              if (FStackInfo.Vars[i].Name = 'Result') then
+                HandleHint('Result not set', Vars[i]._DocPos)
```

Comment:
I think these strings should be defined as resourcestrings/constants in line with exceptions.